scripts/combine/setupCombine.py

```diff
@@ -320,22 +333,55 @@ def assertSample(name, startsWith=["W", "Z"], excludeMatch=[]):
-        return cardTool
-
     cardTool.addSystematic(f"massWeight{label}",
                            processes=signal_samples_forMass,
                            group=f"massShift{label}",
                            noi=not constrainMass,
                            skipEntries=massSkip,
                            mirror=False,
-                           #TODO: Name this
                            noConstraint=not constrainMass,
                            systAxes=["massShift"],
                            passToFakes=passSystToFakes,
     )
 
+    if args.fitMassDiff:
+        suffix = "".join([a.capitalize() for a in args.fitMassDiff.split("-")])
+        mass_diff_args = dict(
+            name=f"massWeight{label}",
+            processes=signal_samples_forMass,
+            rename=f"massDiff{suffix}{label}",
+            group=f"massDiff{label}",
+            systNameReplace=[("Shift",f"Diff{suffix}")],
+            skipEntries=[(m[0].replace("50MeV", "100MeV"),) for m in massSkip],
+            noi=not constrainMass,
+            noConstraint=not constrainMass,
+            mirror=False,
+            systAxes=["massShift"],
+            passToFakes=passSystToFakes,
+        )
+        if args.fitMassDiff == "charge":
+            cardTool.addSystematic(**mass_diff_args,
+                                # # on gen level based on the sample, only possible for mW
+                                # actionMap={m.name: (lambda h, swap=swap_bins: swap(h, "massShift", f"massShift{label}50MeVUp", f"massShift{label}50MeVDown")) 
+                                #     for g in cardTool.procGroups[signal_samples_forMass[0]] for m in cardTool.datagroups.groups[g].members if "minus" in m.name},
+                                # on reco level based on reco charge
+                                actionMap={m.name: (lambda h, swap=hh.swap_histogram_bins: 
+                                        swap(h, "massShift", f"massShift{label}50MeVUp", f"massShift{label}50MeVDown", "charge", 0)) 
+                                    for g in cardTool.procGroups[signal_samples_forMass[0]] for m in cardTool.datagroups.groups[g].members},
+            )
+        elif args.fitMassDiff == "eta-sign":
+            cardTool.addSystematic(**mass_diff_args, 
+                                actionMap={m.name: (lambda h, swap=hh.swap_histogram_bins, s=hist.tag.Slicer(): 
+                                        swap(h, "massShift", f"massShift{label}50MeVUp", f"massShift{label}50MeVDown", "eta", s[0:complex(0,0):]))
+                                    for g in cardTool.procGroups[signal_samples_forMass[0]] for m in cardTool.datagroups.groups[g].members},
+            )
+        elif args.fitMassDiff == "eta-range":
+            cardTool.addSystematic(**mass_diff_args, 
+                                actionMap={m.name: (lambda h, swap=hh.swap_histogram_bins, s=hist.tag.Slicer(): 
```

Comment:
Why do you need to capture the function and the slicer in the lambda?

Reply:
I think it is more save, I'm actually never sure when something has to be captured and when not, so in case of doubt I capture it.

Reply:
It has to be captured if it's changing, e.g., if it's a variable in a loop. In this case I don't see how these could change.

Reply:
thanks, removed the capture